exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitExtension.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.testkit;
+
+import com.exonum.binding.util.LibraryLoader;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+public class TestKitExtension implements ParameterResolver, BeforeEachCallback, AfterEachCallback {
+
+  private static final String KEY = "ResourceKey";
+
+  static {
+    LibraryLoader.load();
+  }
+
+  private final TestKit.Builder testKitBuilder;
+
+  public TestKitExtension(TestKit.Builder testKitBuilder) {
+    this.testKitBuilder = testKitBuilder;
+  }
+
+  @Override
+  public void beforeEach(ExtensionContext extensionContext) {
+    TestKit testKit = testKitBuilder.build();
+    getStore(extensionContext).put(KEY, testKit);
+  }
+
+  @Override
+  public void afterEach(ExtensionContext extensionContext) {
+    TestKit testKit = getStore(extensionContext).get(KEY, TestKit.class);
+    testKit.disposeInternal();
+  }
+
+  @Override
+  public boolean supportsParameter(ParameterContext parameterContext,
+                                   ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+    return parameterContext.getParameter().getType() == TestKit.class;
+  }
+
+  @Override
+  public Object resolveParameter(
+      ParameterContext parameterContext, ExtensionContext extensionContext)
+      throws ParameterResolutionException {
+    return getStore(extensionContext).get(KEY, TestKit.class);
+  }
+
+  private ExtensionContext.Store getStore(ExtensionContext context) {
+    return context.getStore(ExtensionContext.Namespace.create(getClass(),
+        context.getRequiredTestMethod()));
```

Comment:
Is it really needed to include the test method as the namespace? Can't we just create a single namespace as a constant? I think stores of, for instance, test methods running in parallel isolated, and `TempDirectory` in JUnit 5 suggests they are.